### PR TITLE
doc: Fixes wordings in the new advanced_cluster sharding guide.

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

See https://jira.mongodb.org/browse/CLOUDP-269512 for more info

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by jwilliams-mongo — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: docs/guides/advanced-cluster-new-sharding-schema.md [hunk 1]
```diff
@@ -1,23 +1,23 @@
 ---
-page_title: "Migration Guide: Advanced Cluster New Sharding Schema"
+page_title: "Migration Guide: Advanced Cluster New Sharding Configurations"
 ---
 
-# Migration Guide: Advanced Cluster New Sharding Schema
+# Migration Guide: Advanced Cluster New Sharding Configurations
```
Comment (AgustinBettati) [inline docs/guides/advanced-cluster-new-sharding-schema.md:5]: Seeing that in advanced_cluster resource and data source docs we also use `sharding schema` for some example titles, would be great if we can adjust for consistency on this change.

File: docs/guides/advanced-cluster-new-sharding-schema.md [hunk 2]
```diff
@@ -277,7 +277,7 @@ resource "mongodbatlas_advanced_cluster" "test" {
 <a id="use-iss"></a>
 ### Use Independent Shard Scaling 
 
-Use the new sharding schema. Each shard must be represented with a unique replication_specs element and `num_shards` must not be used, as illustrated in the following example.
+Use the new sharding configurations. Each shard must be represented with a unique replication_specs element and `num_shards` must not be used, as illustrated in the following example.
```
Comment (jwilliams-mongo) [inline docs/guides/advanced-cluster-new-sharding-schema.md]: drive-by nit
```suggestion
Use the new sharding configurations. Each shard must be represented with a unique `replication_specs` element and `num_shards` must not be used, as illustrated in the following example:
```

**Review by jwilliams-mongo — APPROVED** [review]

File: docs/guides/advanced-cluster-new-sharding-schema.md (hunk 1 above)
Comment (marcosuma) [reply docs/guides/advanced-cluster-new-sharding-schema.md:5]: changes extended to other files

**Review by jwilliams-mongo — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]
LGTM
